WordPress/Classes/Services/BlockEditorCache.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+import CryptoKit
+import WordPressShared
+import WordPressData
+
+final class BlockEditorCache {
+    static let shared = BlockEditorCache()
+
+    private let rootURL: URL
+    private let blockSettingsURL: URL
+
+    private init() {
+        rootURL = URL.cachesDirectory
+            .appendingPathComponent("GutenbergKit", isDirectory: true)
+        blockSettingsURL = rootURL
+            .appendingPathComponent("BlockSettings", isDirectory: true)
+
+        // Create directory if it doesn't exist
+        try? FileManager.default.createDirectory(at: blockSettingsURL, withIntermediateDirectories: true)
+    }
+
+    // MARK: - Block Settings
+
+    func saveBlockSettings(_ settings: [String: Any], for blogID: TaggedManagedObjectID<Blog>) {
+        let fileURL = makeBlockSettingsURL(for: blogID)
+
+        do {
+            let data = try JSONSerialization.data(withJSONObject: settings, options: [.prettyPrinted])
+            try data.write(to: fileURL)
+        } catch {
+            DDLogError("Failed to save block editor settings: \(error)")
+        }
+    }
+
+    func getBlockSettings(for blogID: TaggedManagedObjectID<Blog>) -> [String: Any]? {
+        let fileURL = makeBlockSettingsURL(for: blogID)
+
+        guard FileManager.default.fileExists(atPath: fileURL.path) else {
+            return nil
+        }
+
+        do {
+            let data = try Data(contentsOf: fileURL)
+            let object = try JSONSerialization.jsonObject(with: data, options: [])
+            return object as? [String: Any]
+        } catch {
+            DDLogError("Failed to load block editor settings: \(error)")
+            // If the file is corrupted, delete it
+            try? FileManager.default.removeItem(at: fileURL)
+            return nil
+        }
+    }
+
+    func deleteBlockSettings(for blogID: TaggedManagedObjectID<Blog>) {
+        let fileURL = makeBlockSettingsURL(for: blogID)
+
+        do {
+            try FileManager.default.removeItem(at: fileURL)
+        } catch {
+            DDLogError("Failed to delete block editor settings: \(error)")
+        }
+    }
+
+    private func makeBlockSettingsURL(for blogID: TaggedManagedObjectID<Blog>) -> URL {
+        let key = sha256(objectID: blogID.objectID)
+        return blockSettingsURL.appendingPathComponent("\(key).json")
+    }
+
+    private func sha256(objectID: NSManagedObjectID) -> String {
+        let uriString = objectID.uriRepresentation().absoluteString
```

Comment:
`NSManagedObjectID` seems to be the best way to uniquely identity a blog of any type: self-hosted or otherwise. `SHA256` ensure that it contains only characters valid in a filename. Drawback: harder to match blog to file if you are manually inspecting the files in the filesystem.